### ISSUE TITLE
KILL QUERY won't finish

### ISSUE THREAD

**Issue by simPod** [yandex/ClickHouse#3416]
I have a query running that blocks some other queries in queue. 

I tried to kill it with 

```sql
KILL QUERY where query_id = 'f7373965-0605-45a2-878e-ed158bf88e64' SYNC;
```

However, the query seems it's blocked by something as well and won't finish. It hangs on

```
→ Progress: 40.00 rows, 64.52 KB (391.03 rows/s., 630.72 KB/s.)  99%Cannot load data for command line suggestions: Code: 209, e.displayText() = DB::NetException: Timeout exceeded while reading from socket (127.0.0.1:9000): while receiving packet from localhost:9000, 127.0.0.1, e.what() = DB::NetException
```

Is there any other way to force kill it?

**Comment by filimonov** [comment]
Duplicate of https://github.com/yandex/ClickHouse/issues/1576 and #3964
